### PR TITLE
Add Komga LXC

### PR DESCRIPTION
**Pull request by yry-dev**
## Description

Add new script for a komga lxc.  This is my first attempt at one so candid feedback is welcomed.
[Komga](https://komga.org/) is a media server for your comics, mangas, BDs, magazines and eBooks.  Kinda like plex for ebooks and comics.

## Type of change
Please check the relevant option(s):

- [ ] Bug fix (non-breaking change that resolves an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (a fix or feature that would cause existing functionality to change unexpectedly)
- [x] New script (a fully functional and thoroughly tested script or set of scripts.)

## Prerequisites
The following efforts must be made for the PR to be considered. Please check when completed:
- [x] Self-review performed (I have reviewed my code, ensuring it follows established patterns and conventions)
- [x] Testing performed (I have tested my changes, ensuring everything works as expected)
- [x] Documentation updated (I have updated any relevant documentation)

### CODE REVIEW

File: ct/komga.sh [hunk 1]
```diff
@@ -0,0 +1,78 @@
+#!/usr/bin/env bash
+source <(curl -s https://raw.githubusercontent.com/community-scripts/ProxmoxVE/main/misc/build.func)
+# Copyright (c) 2024 madelyn
```
Comment (MickLesk) [inline ct/komga.sh:3]: Copyright (c) 2021-2024 community-scripts ORG

File: ct/komga.sh [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+  NS=""
+  MAC=""
+  VLAN=""
+  SSH="no"
+  VERB="no"
+  echo_default
+}
+
+function update_script() {
+header_info
```
Comment (MickLesk) [inline ct/komga.sh:56]: add here:
header_info (already here)
check_container_storage
check_container_resources

File: ct/komga.sh [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+if [[ ! -d /opt/komga ]]; then msg_error "No ${APP} Installation Found!"; exit; fi
+msg_info "Updating ${APP}"
+RELEASE=$(curl -s https://api.github.com/repos/gotson/komga/releases/latest | grep "tag_name" | awk '{print substr($2, 2, length($2)-3) }')
+if [[ ! -d /opt/komga/komga-${RELEASE}.jar ]]; then
```
Comment (MickLesk) [inline ct/komga.sh:60]: if [[ ! -f | not -d?

File: ct/komga.sh [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+  systemctl stop komga
```
Comment (MickLesk) [inline ct/komga.sh:61]: msg_info "Stopping ${APP}"

msg_ok "Stopped ${APP}"

File: ct/komga.sh [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+  SSH="no"
+  VERB="no"
+  echo_default
+}
+
+function update_script() {
+header_info
+if [[ ! -d /opt/komga ]]; then msg_error "No ${APP} Installation Found!"; exit; fi
+msg_info "Updating ${APP}"
+RELEASE=$(curl -s https://api.github.com/repos/gotson/komga/releases/latest | grep "tag_name" | awk '{print substr($2, 2, length($2)-3) }')
```
Comment (MickLesk) [inline ct/komga.sh:59]: Add after RELEASE:
if [[ ! -f /opt/${APP}_version.txt ]] || [[ "${RELEASE}" != "$(cat /opt/${APP}_version.txt)" ]]; then

File: ct/komga.sh [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+if [[ ! -d /opt/komga/komga-${RELEASE}.jar ]]; then
+  systemctl stop komga
+  msg_info "Downloading ${APP} v$RELEASE"
+  wget -q "https://github.com/gotson/komga/releases/download/v$RELEASE/komga-${RELEASE}.jar"
+  mkdir -p /opt/komga
```
Comment (MickLesk) [inline ct/komga.sh:64]: not needed for update

File: ct/komga.sh [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+}
+
+function update_script() {
+header_info
+if [[ ! -d /opt/komga ]]; then msg_error "No ${APP} Installation Found!"; exit; fi
+msg_info "Updating ${APP}"
+RELEASE=$(curl -s https://api.github.com/repos/gotson/komga/releases/latest | grep "tag_name" | awk '{print substr($2, 2, length($2)-3) }')
+if [[ ! -d /opt/komga/komga-${RELEASE}.jar ]]; then
+  systemctl stop komga
+  msg_info "Downloading ${APP} v$RELEASE"
```
Comment (MickLesk) [inline ct/komga.sh:62]: Updating not Downloading

File: ct/komga.sh [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+  wget -q "https://github.com/gotson/komga/releases/download/v$RELEASE/komga-${RELEASE}.jar"
+  mkdir -p /opt/komga
+  mv -f komga-${RELEASE}.jar /opt/komga/komga-${RELEASE}.jar
+  systemctl start komga
```
Comment (MickLesk) [inline ct/komga.sh:66]: msg_info "Starting ${APP}"
  systemctl start komga
  msg_ok "Started ${APP}"

File: install/komga-install.sh [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+source /dev/stdin <<< "$FUNCTIONS_FILE_PATH"
+color
+verb_ip6
+catch_errors
+setting_up_container
+network_check
+update_os
+
+msg_info "Installing Dependencies"
+$STD apt-get install -y curl
```
Comment (MickLesk) [inline install/komga-install.sh:17]: core dependencies missing (mc, sudo)
$STD apt-get install -y \
  curl \
  mc \
  sudo \
  openjdk-17-jre

File: install/komga-install.sh [hunk 10]
```diff
@@ -0,0 +1,55 @@
...
+$STD apt-get install -y openjdk-17-jre
+msg_ok "Installed Dependencies"
+
+RELEASE=$(curl -s https://api.github.com/repos/gotson/komga/releases/latest | grep "tag_name" | awk '{print substr($2, 2, length($2)-3) }')
+
+msg_info "Installing Komga"
+wget -q https://github.com/gotson/komga/releases/download/${RELEASE}/komga-${RELEASE}.jar
+mkdir -p /opt/komga
+mv -f komga-${RELEASE}.jar /opt/komga/komga-${RELEASE}.jar
+msg_ok "Installed Komga 😊"
```
Comment (MickLesk) [inline install/komga-install.sh:27]: Without Emoji ;-)

Comment (MickLesk) [reply install/komga-install.sh:27]: and add before: 
echo "${RELEASE}" >"/opt/${APPLICATION}_version.txt"

File: install/komga-install.sh [hunk 11]
```diff
@@ -0,0 +1,55 @@
...
+
+msg_info "Creating Service"
+service_path="/etc/systemd/system/komga.service"
```
Comment (MickLesk) [inline install/komga-install.sh:30]: does not match the existing conversations

msg_info "Creating Service"
cat <<EOF >/etc/systemd/system/komga.service"
...
...
...
...
EOF

**Review by MickLesk — COMMENTED** [review]
Checkout my comments, there are some things todo :smile:

**Comment by DysfunctionalProgramming** [comment]
> Checkout my comments, there are some things todo 😄

OK I think I took care of all the change requests.

File: ct/komga.sh [hunk 12]
```diff
@@ -21,7 +21,7 @@ check_container_storage
 check_container_resources
 echo -e "Loading..."
 APP="Komga"
-var_disk="2"
+var_disk="4"
```
Comment (DysfunctionalProgramming) [inline ct/komga.sh:24]: If your changing this back to 4 do we need to update the komga.json to reflect that?

Comment (MickLesk) [reply ct/komga.sh:24]: Yes, 2 is a little bit less, the storage check says no :D

**Comment by MickLesk** [comment]
@havardthom can you take a look and give Feedback? I have made some improvements, but only from my cell phone, I think I may have overlooked or forgotten something

File: ct/komga.sh [hunk 13]
```diff
@@ -0,0 +1,92 @@
...
+source <(curl -s https://raw.githubusercontent.com/community-scripts/ProxmoxVE/main/misc/build.func)
+# Copyright (c) 2021-2024 community-scripts ORG
+# Author: madelyn (DysfunctionalProgramming)
+# License: MIT
+# https://github.com/community-scripts/ProxmoxVE/raw/main/LICENSE
+
+function header_info {
+clear
+cat <<"EOF"
+ _  __
```
Comment (havardthom) [inline ct/komga.sh:11]: Wrong ascii, use this https://www.patorjk.com/software/taag/#p=display&f=Slant&t=Komga

File: ct/komga.sh [hunk 14]
```diff
@@ -0,0 +1,92 @@
...
+ _  __                           
+| |/ /___  _ __ ___   __ _  __ _ 
+| ' // _ \| '_ ` _ \ / _` |/ _` |
+| . \ (_) | | | | | | (_| | (_| |
+|_|\_\___/|_| |_| |_|\__, |\__,_|
+                     |___/       
+EOF
+}
+header_info
+check_container_storage
```
Comment (havardthom) [inline ct/komga.sh:20]:
```suggestion
```

File: ct/komga.sh [hunk 15]
```diff
@@ -0,0 +1,92 @@
...
+check_container_resources
```
Comment (havardthom) [inline ct/komga.sh:21]:
```suggestion
```

File: ct/komga.sh [hunk 16]
```diff
@@ -0,0 +1,92 @@
...
+  mv -f komga-${RELEASE}.jar /opt/komga/komga.jar
+  echo "${RELEASE}" >/opt/${APP}_version.txt
+  msg_ok "Updated ${APP} to ${RELEASE}"
+
+  msg_info "Starting ${APP}"
+  systemctl start komga
+  msg_ok "Started ${APP}"
+  msg_ok "Updated Successfully"
+else
+  msg_ok "No update required. ${APP} is already at v${RELEASE}."
```
Comment (havardthom) [inline ct/komga.sh:81]:
```suggestion
  msg_ok "No update required. ${APP} is already at ${RELEASE}."  
```

File: install/komga-install.sh [hunk 17]
```diff
@@ -0,0 +1,57 @@
...
+color
+verb_ip6
+catch_errors
+setting_up_container
+network_check
+update_os
+
+msg_info "Installing Dependencies"
+$STD apt-get install -y \
+curl \
```
Comment (havardthom) [inline install/komga-install.sh:18]:
```suggestion
  curl \
```

File: install/komga-install.sh [hunk 18]
```diff
@@ -0,0 +1,57 @@
...
+mc \
```
Comment (havardthom) [inline install/komga-install.sh:19]:
```suggestion
  mc \
```

File: install/komga-install.sh [hunk 19]
```diff
@@ -0,0 +1,57 @@
...
+sudo \
```
Comment (havardthom) [inline install/komga-install.sh:20]:
```suggestion
  sudo \
```

File: install/komga-install.sh [hunk 20]
```diff
@@ -0,0 +1,57 @@
...
+openjdk-17-jre
```
Comment (havardthom) [inline install/komga-install.sh:21]:
```suggestion
  openjdk-17-jre
```

File: install/komga-install.sh [hunk 21]
```diff
@@ -0,0 +1,57 @@
...
+mv -f komga-${RELEASE}.jar /opt/komga/komga.jar
+echo "${RELEASE}" >"/opt/${APPLICATION}_version.txt"
+msg_ok "Installed Komga"
+
+msg_info "Creating Service"
+cat <<EOF >/etc/systemd/system/komga.service
+[Unit]
+Description=Komga
+After=syslog.target network.target
+[Service]
```
Comment (havardthom) [inline install/komga-install.sh:37]:
```suggestion

[Service]
```

File: install/komga-install.sh [hunk 22]
```diff
@@ -0,0 +1,57 @@
...
+UMask=0002
+Type=simple
+WorkingDirectory=/opt/komga/
+ExecStart=java -jar -Xmx2g komga.jar
```
Comment (havardthom) [inline install/komga-install.sh:41]:
```suggestion
ExecStart=/usr/bin/java -jar -Xmx2g komga.jar
```

File: install/komga-install.sh [hunk 23]
```diff
@@ -0,0 +1,57 @@
...
+echo "${RELEASE}" >"/opt/${APPLICATION}_version.txt"
+msg_ok "Installed Komga"
+
+msg_info "Creating Service"
+cat <<EOF >/etc/systemd/system/komga.service
+[Unit]
+Description=Komga
+After=syslog.target network.target
+[Service]
+UMask=0002
```
Comment (havardthom) [inline install/komga-install.sh:38]: Is this actually needed or is it copied?

File: install/komga-install.sh [hunk 24]
```diff
@@ -0,0 +1,57 @@
...
+Type=simple
+WorkingDirectory=/opt/komga/
+ExecStart=java -jar -Xmx2g komga.jar
+TimeoutStopSec=20
+KillMode=process
+Restart=on-failure
+[Install]
```
Comment (havardthom) [inline install/komga-install.sh:45]:
```suggestion

[Install]
```

File: json/komga.json [hunk 25]
```diff
@@ -0,0 +1,34 @@
...
+    "slug": "komga",
+    "categories": [
+        12
+    ],
+    "date_created": "2024-11-15",
+    "type": "ct",
+    "updateable": true,
+    "privileged": false,
+    "interface_port": "25600",
+    "documentation": null,
```
Comment (havardthom) [inline json/komga.json:12]:
```suggestion
    "documentation": "https://komga.org/docs/introduction",
```

File: json/komga.json [hunk 26]
```diff
@@ -0,0 +1,34 @@
...
+    "website": "https://komga.org/",
+    "logo": "https://github.com/gotson/komga/raw/master/.github/readme-images/app-icon.png",
```
Comment (havardthom) [inline json/komga.json:14]: This icon has a white background, have you tested how it looks on the website?

**Review by havardthom — DISMISSED** [review]

File: json/komga.json (hunk 26 above)
Comment (havardthom) [reply json/komga.json:14]: nvm I can see it in the readme, guess it's just the browser adding white background when I open it

File: install/komga-install.sh (hunk 22 above)
Comment (MickLesk) [reply install/komga-install.sh:41]: Is the Name of the jar File really konga.jar ?

File: ct/komga.sh (hunk 15 above)
Comment (MickLesk) [reply ct/komga.sh:21]: Why? Its the new function

File: ct/komga.sh (hunk 14 above)
Comment (MickLesk) [reply ct/komga.sh:20]: Why? Its the new function

Comment (havardthom) [reply ct/komga.sh:20]: It is a duplicate line and in the wrong place, it should only be in update function

File: ct/komga.sh (hunk 15 above)
Comment (havardthom) [reply ct/komga.sh:21]: It is a duplicate line and in the wrong place, it should only be in update function

**Review by MickLesk — APPROVED** [review]

**Review by havardthom — APPROVED** [review]

**Pull request merged**